addon/lib/survey.js

```diff
@@ -0,0 +1,146 @@
+/*
+ *   This Source Code is subject to the terms of the Mozilla Public License
+ *   version 2.0 (the 'License'). You can obtain a copy of the License at
+ *   http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ *   Originally Liberated from
+ *   https://github.com/lmorchard/fireriver/blob/master/lib/main.js#L369
+ *   & https://dxr.mozilla.org/mozilla-central/source/browser/components/uitour/UITour.jsm
+ */
+const {Cc, Ci} = require('chrome');
+const {setTimeout} = require('sdk/timers');
+const tabs = require('sdk/tabs');
+const querystring = require('sdk/querystring');
+const simplePrefs = require('sdk/simple-prefs');
+const store = require('sdk/simple-storage').storage;
+
+const NUM_STARS = simplePrefs.prefs.NUM_STARS;
+
+module.exports = {launchSurvey: launchSurvey};
+
+function getAnonEl(win, box, attrName) {
+  return win.document.getAnonymousElementByAttribute(box, 'anonid', attrName);
+}
+
+// set timecheck flags for initial run.
+if (store.surveyChecks === undefined) store.surveyChecks = {};
+['twoDaysSent', 'oneWeekSent', 'threeWeeksSent', 'monthAndAHalfSent'].forEach(k => {
+  if (store.surveyChecks[k] === undefined) store.surveyChecks[k] = {};
+});
+
+// check if the addon is due for a survey at the current interval
+function checkInstallDate(installDate, addonId) {
+  const dateDiff = new Date().getDate() - new Date(installDate).getDate();
+
+  if (!(addonId in store.twoDaysSent) && dateDiff >= 2) {
```

Comment:
Getting errors, here, when I try running this stuff. Maybe these should be `store.surveyChecks['twoDaysSent']` rather than `store.twoDaysSent`, etc?